bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/inst/ProbeArmGeometry.scala

```diff
@@ -0,0 +1,109 @@
+   */
+  def armAdjustment(ctx: ObsContext, guideStarCoords: Coordinates, offset: Offset): Option[ArmAdjustment]
+
+  def armAdjustment(ctx: ObsContext, guideStar: SPTarget, offset: Offset): Option[ArmAdjustment] =
+    armAdjustment(ctx, guideStar.getTarget.getSkycalcCoordinates.toNewModel, offset)
+
+  def armAdjustment(ctx: ObsContext, guideStar: SiderealTarget, offset: Offset): Option[ArmAdjustment] =
+    armAdjustment(ctx, guideStar.coordinates, offset)
+
+  def armAdjustmentAsJava(ctx: ObsContext, guideStarCoords: Coordinates, offset: Offset): GOption[ArmAdjustment] =
+    armAdjustment(ctx, guideStarCoords, offset).asGeminiOpt
+
+  def armAdjustmentAsJava(ctx: ObsContext, guideStar: SPTarget, offset: Offset): GOption[ArmAdjustment] =
+    armAdjustment(ctx, guideStar, offset).asGeminiOpt
+
+  def armAdjustmentAsJava(ctx: ObsContext, guideStar: SiderealTarget, offset: Offset): GOption[ArmAdjustment] =
+    armAdjustment(ctx, guideStar, offset).asGeminiOpt
+
+  /**
+   * For a given context and offset, calculate the arm adjustment that will be used for the primary / selected guide
+   * star.
+   * @param ctx    the context representing the configuration
+   * @param offset the offset for which to calculate the adjustment
+   * @return       the probe arm adjustments for this data
+   */
+  def armAdjustment(ctx: ObsContext, offset: Offset): Option[ArmAdjustment] =
+    (for {
+      c            <- Option(ctx)
+      guideTargets <- ctx.getTargets.getPrimaryGuideProbeTargets(guideProbeInstance).asScalaOpt
+      guideStar    <- guideTargets.getPrimary.asScalaOpt
+    } yield armAdjustment(ctx, guideStar, offset)).flatten
+
+  def armAdjustmentAsJava(ctx: ObsContext, offset: Offset): GOption[ArmAdjustment] =
+    armAdjustment(ctx, offset).asGeminiOpt
+}
+
+object ProbeArmGeometry {
+  private lazy val maxArcsecs = 360 * 60 * 60d
+
+  /**
+   * Convert a value in arcseconds (i.e. in the range [0,maxArcsecs) to its canonical value, i.e.
+   * the equivalent value in the range [-maxArcsecs/2,maxArcsecs/2).
+   */
+  private implicit class CanonicalValue(val v: Double) extends AnyVal {
+    def toCanonicalValue: Double = {
+      val v1 = math.IEEEremainder(v, maxArcsecs)
+      val v2 = v1 - maxArcsecs
+      if (math.abs(v1) <= math.abs(v2)) v1 else v2
+    }
+  }
+
+  /**
+   * Convert a point representing coordinates in arcsec to canonical value in both coordinates.
+   */
+  implicit class CanonicalPoint(val p: Point2D) extends AnyVal {
+    def toCanonicalForm: Point2D =
+      new Point2D.Double(p.getX.toCanonicalValue, p.getY.toCanonicalValue)
+  }
+}
+
+/**
+ * A representation of the adjustment made to the default list of shapes when using a specified guide star.
+ * @param angle           the angle which will be used by the probe arm
+ * @param guideStarCoords the coordinates (in arcsec) where the probe arm will be placed
+ */
+case class ArmAdjustment(angle: Double, guideStarCoords: Point2D)
```

Comment:
Would it be a good idea to represent this with `edu.gemini.spModel.core.Angle` instead of a raw `Double`?  A `Double` value can be anything but an `Angle.toDegrees` will always give you a value [0, 360).